rls doesn't build

**Describe the bug**
A clear and concise description of what the bug is.

on master (254d6e5cbf5d366df46d4651706f9ff59427a57d)

```
   Compiling rls-rustc v0.6.0 (/build/rustc-1.40.0-src/src/tools/rls/rls-rustc)
error[E0463]: can't find crate for `rustc`
 --> src/tools/rls/rls-rustc/src/lib.rs:4:1
  |
4 | extern crate rustc;
  | ^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `rls-rustc`.
warning: build failed, waiting for other jobs to finish...
error: build failed
builder for '/nix/store/bbjm5mgxyfmxj72ypnv7x65588icm9hh-rls-1.40.0.drv' failed with exit code 101
error: build of '/nix/store/bbjm5mgxyfmxj72ypnv7x65588icm9hh-rls-1.40.0.drv' failed
```

**To Reproduce**
Steps to reproduce the behavior:
1. cd nixpkgs
2. nix-build -A rls

**Expected behavior**

**Screenshots**

**Additional context**

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.92, NixOS, 20.03pre208171.2e8fc97dbfa (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.1`
 - channels(root): `"nixos-20.03pre208171.2e8fc97dbfa"`
 - channels(bbigras2): `"home-manager, nixos-19.09-19.09.1776.b926503738c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/compilers/rust/rustc.nix
+++ b/pkgs/development/compilers/rust/rustc.nix
@@ -132,9 +132,14 @@ in stdenv.mkDerivation rec {
   outputs = [ "out" "man" "doc" ];
   setOutputFlags = false;
 
-  # remove references to llvm-config in lib/rustlib/x86_64-unknown-linux-gnu/codegen-backends/librustc_codegen_llvm-llvm.so
-  # and thus a transitive dependency on ncurses
   postInstall = ''
+    # install rustc-dev components. Necessary to build rls, clippy...
+    python x.py dist rustc-dev
+    tar xf build/dist/rustc-dev*tar.gz
+    cp -r rustc-dev*/rustc-dev*/lib/* $out/lib/
+
+    # remove references to llvm-config in lib/rustlib/x86_64-unknown-linux-gnu/codegen-backends/librustc_codegen_llvm-llvm.so
+    # and thus a transitive dependency on ncurses
     find $out/lib -name "*.so" -type f -exec remove-references-to -t ${llvmShared} '{}' '+'
   '';
 

